pkg/proxy/iptables/proxier.go

```diff
@@ -1066,56 +1076,73 @@ func (proxier *Proxier) syncProxyRules() {
 			activeNATChains[policyClusterChain] = true
 		}
 
-		if hasEndpoints && svcInfo.ExternallyAccessible() && svcInfo.NodeLocalExternal() {
-			if chain, ok := existingNATChains[svcXlbChain]; ok {
+		// Declare the policyLocalChain if needed.
+		if hasEndpoints && svcInfo.UsesLocalEndpoints() {
+			if chain, ok := existingNATChains[policyLocalChain]; ok {
+				proxier.natChains.WriteBytes(chain)
+			} else {
+				proxier.natChains.Write(utiliptables.MakeChainLine(policyLocalChain))
+			}
+			activeNATChains[policyLocalChain] = true
+		}
+
+		// If any "external" destinations are enabled, set up external traffic
+		// handling.  All captured traffic for all external destinations should
+		// jump to externalTrafficChain, which will handle some special-cases
+		// and then jump to externalPolicyChain.
+		if hasEndpoints && svcInfo.ExternallyAccessible() {
+			if chain, ok := existingNATChains[externalTrafficChain]; ok {
 				proxier.natChains.WriteBytes(chain)
 			} else {
-				proxier.natChains.Write(utiliptables.MakeChainLine(svcXlbChain))
+				proxier.natChains.Write(utiliptables.MakeChainLine(externalTrafficChain))
 			}
-			activeNATChains[svcXlbChain] = true
+			activeNATChains[externalTrafficChain] = true
 
-			// The XLB chain redirects all pod -> external VIP
-			// traffic to the Service's ClusterIP instead. This happens
-			// whether or not we have local endpoints; only if localDetector
-			// is implemented
-			if proxier.localDetector.IsImplemented() {
+			if !svcInfo.NodeLocalExternal() || !proxier.localDetector.IsImplemented() {
+				// If we are proxying globally (or if we can't tell local from
```

Comment:
Maybe fix up this comment? "Proxying globally" is outdated terminology. (Likewise "proxying locally" below.)

Reply:
ACK